persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/JdbcBasePersistenceImpl.java

```diff
@@ -773,6 +773,49 @@ public PolarisPrincipalSecrets generateNewPrincipalSecrets(
     return principalSecrets;
   }
 
+  @Nullable
+  @Override
+  public PolarisPrincipalSecrets resetPrincipalSecrets(
+      @Nonnull PolarisCallContext callCtx,
+      @Nonnull String clientId,
+      long principalId,
+      String customClientId,
+      String customClientSecret) {
+
+    PolarisPrincipalSecrets principalSecrets;
+    if (customClientId == null) {
+      principalSecrets = new PolarisPrincipalSecrets(principalId, clientId, customClientSecret);
```

Comment:
I believe `customClientSecret` may still be `null` here (when the REST request does not have it). Should be generate a random secret when it is not provided by the user?

Reply:
My mistake - I missed that change :facepalm: All good here.